apisix/plugin.lua

```diff
@@ -849,6 +850,71 @@ check_plugin_metadata = function(item)
 end
 
 
+local enable_data_encryption
+local function enable_gde()
+    if enable_data_encryption == nil then
+        enable_data_encryption =
+            core.table.try_read_attr(local_conf, "apisix", "data_encryption", "enable")
+        _M.enable_data_encryption = enable_data_encryption
```

Comment:
The cache won't take effect when `core.table.try_read_attr(local_conf, "apisix", "data_encryption", "enable")` is nil. is this by design?

Reply:
+1

Reply:
from #8407

```
apisix:
    data_encryption: 
        enable: false               # If not set, the default value is `false`.
```

if `apisix.data_encryption.enable` is nil, it is the same as not set `apisix.data_encryption.enable`, then `_M.enable_data_encryption` would be `false`. 

`_M.enable_data_encryption` is used by `if method == "get" and plugin.enable_data_encryption then` in `apisix/admin/init.lua`, it won't into `utils.decrypt_params`

Reply:
@tzssangglass 
`enable_data_encryption` would be nil in this case? So `enable_data_encryption == nil` will always get a true result and this function doesn't provide a cache if the user doesn't provide a configuration.

Reply:
1. if the user doesn't config `apisix.data_encryption.enable`, then `core.table.try_read_attr(local_conf, "apisix", "data_encryption", "enable")` would get the default value `false` from `conf/config-default.yaml`
2. `enable_data_encryption` would be set to `false` as module var.
3. in the next time, `enable_data_encryption == nil` would get false